### ISSUE TITLE
Segmentation fault and corrupted output during inference with large 3D U-Net

### ISSUE THREAD

**Issue by pzharrington** [tensorflow/tensorflow#34841]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): YES
- OS Platform and Distribution: Linux CLE 7.0UP00
- TensorFlow installed from (source or binary): binary
- TensorFlow version: 2.0.0
- Python version: 3.7.4

**Describe the current behavior**
I have a 3D U-Net model that I am trying to use for very large inputs (arrays as large as 1024^3). Obviously, I cannot fit this in GPU memory so I am running the network on CPU compute nodes which have a very large amount of RAM available (~350 GB). I have a "bare-bones" script which performs only the forward pass using the 3D convolution methods from `tf.nn`, but I am only able to process arrays of size  <= 640^3 without issues. If I go for the next feasible size (which is 704^3 since I can only increase spatial dimensions of the input by multiples of 64), the following behavior occurs:

* If using tensorflow with the Intel MKL library, which is optimized for running on CPUs on my system, the network runs without crashing but the generated output is corrupted (see attached images). The corruption seems to only occur along the outermost spatial axis of the output array.

* If using tensorflow-gpu or regular tensorflow (without Intel MKL), the program crashes with a segmentation fault

Though the input sizes are very large, I would expect tensorflow to be able to complete the computations without issue since I have enough RAM available. The difference in behavior with or without Intel MKL is also perplexing.

**Code to reproduce the issue**
Below is a script that performs the U-Net computation on random data and reproduces the segfault/bug:
```
import tensorflow as tf
import numpy as np

size=704
dtype = np.float32
sample = np.random.normal(size=(1,size,size,size,4)).astype(dtype)
print(sample.shape)

convs = [(4, 4, 4, 4, 64),
         (4, 4, 4, 64, 128),
         (4, 4, 4, 128, 256),
         (4, 4, 4, 256, 512),
         (4, 4, 4, 512, 512),
         (4, 4, 4, 512, 512)]
Tconvs = [(4, 4, 4, 512, 512),
          (4, 4, 4, 512, 1024),
          (4, 4, 4, 256, 1024),
          (4, 4, 4, 128, 512),
          (4, 4, 4, 64, 256),
          (4, 4, 4, 5, 128)]

x = tf.constant(sample)
skips = []
shapes = []

# Downsampling
for kernelshape in convs:
    bias = np.zeros(shape=(kernelshape[-1]))
    kernel = np.random.normal(loc=0.0, scale=0.02, size=kernelshape)
    x = tf.nn.conv3d(x, filters=kernel.astype(dtype), strides=[1,2,2,2,1], padding='SAME')
    x = tf.nn.bias_add(x, bias.astype(dtype))
    x = tf.nn.leaky_relu(x, alpha=0.3)
    skips.append(x)
    shapes.append(x.shape)
    print(x.shape)

# Upsampling with skip connections
skips = list(reversed(skips[:-1]))
for kernelshape, skip in zip(Tconvs, skips):
    bias = np.zeros(shape=(kernelshape[3]))
    kernel = np.random.normal(loc=0.0, scale=0.02, size=kernelshape)
    out_shape = np.array(x.shape.as_list())
    out_shape[1:4] = 2*out_shape[1:4]
    out_shape[-1] = kernel.shape[3]
    x = tf.nn.conv3d_transpose(x, filters=kernel.astype(dtype), strides=2, padding='SAME', output_shape=out_shape)
    x = tf.nn.bias_add(x, bias.astype(dtype))
    x = tf.nn.relu(x)
    x = tf.concat([x, skip], axis=-1)
    print(x.shape)

# Final upsampling conv
kernelshape = Tconvs[-1]
bias = np.zeros(shape=(kernelshape[3]))
kernel = np.random.normal(loc=0.0, scale=0.02, size=kernelshape)
out_shape = np.array(x.shape.as_list())
out_shape[1:4] = 2*out_shape[1:4]
out_shape[-1] = kernel.shape[3]
x = tf.nn.conv3d_transpose(x, filters=kernel.astype(dtype), strides=2, padding='SAME', output_shape=out_shape)
x = tf.nn.bias_add(x, bias.astype(dtype))
x = tf.math.tanh(x)

print(x.shape)
x = x.numpy().astype(np.float32)
```

**Other info / logs**
The attached images show the output of the U-Net (running on random noise) for the problematic size (704^3), via 2D slices along the x-y and x-z planes: 
![image](https://user-images.githubusercontent.com/48932392/70189520-613e2180-16a8-11ea-970f-27bdd58546f1.png)
The images should look like uniform noise, but instead have a sharp change occur along the x-axis after index~520.

On my actual dataset, the corruption looks like: 
![image](https://user-images.githubusercontent.com/48932392/70190128-fc83c680-16a9-11ea-9fb1-984be9673381.png)

**Comment by penpornk** [comment]
Thank you for reporting the issue! 

I tried this on an Ubuntu 16.04 VM + Python 3.7.4 + 360GB memory. I could reproduce the corrupted image issue with both TensorFlow-MKL (Intel-optmized TensorFlow) and regular TensorFlow. (Regular TensorFlow didn't crash for me.) Will try to fix the corrupted image issue first and then we can see if regular TensorFlow still crash for you.

**Comment by mohantym** [comment]
@pzharrington @penpornk  !
The session is crashing at below line in Colab .
sample = np.random.normal(size=(1,size,size,size,4)).astype(dtype)
before going to other steps in 2.8 version onwards.

Attached gist [2.10.0 ](https://colab.sandbox.google.com/gist/mohantym/9cc71719907fd5efe0b5870a52d69391/git_34841.ipynb#scrollTo=ue1iyKy9swWf)for reference.

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
